standards/core/lightpush.md

```diff
@@ -0,0 +1,88 @@
+---
+slug: 19
+title: 19/WAKU2-LIGHTPUSH
+name: Waku v2 Light Push
+status: draft
+editor: Zoltan Nagy <zoltan@status.im> 
+contributors: 
+  - Hanno Cornelius <hanno@status.im>
+  - Daniel Kaiser <danielkaiser@status.im>
+  - Oskar Thorén <oskarth@titanproxy.com>
+---
+
+previous version: `/vac/waku/lightpush/2.0.0-beta1` [19/WAKU2-LIGHTPUSH](https://rfc.vac.dev/waku/standards/core/19/lightpush)
+
+---
+**Protocol identifier**: `/vac/waku/lightpush/2.0.0-beta2`
+
+## Motivation and Goals
+
+Light nodes with short connection windows and limited bandwidth wish to publish messages into the Waku network.
+Additionally, there is sometimes a need for confirmation that a message has been received "by the network"
+(here, at least one node).
+
+`19/WAKU2-LIGHTPUSH` is a request/response protocol for this.
+
+## Payloads
+
+```protobuf
+syntax = "proto3";
+
+message LightPushRequest {
+    enum RequestType { // Future extension point
+        RELAY = 0;
+    }
+    
+    string request_id = 1;
+    RequestType kind = 10;  // default is RELAY, placeholder for future extensions
+    string pubsub_topic = 20;
+    WakuMessage message = 21;
+}
+
+message LightPushResponse {
+    enum Status {
+        SUCCESS = 0;
+        BAD_REQUEST = 400;
+        NO_PEERS_TO_RELAY = 404;
+        PAYLOAD_TOO_LARGE = 413;
+        UNSUPPORTED_TOPIC = 415;
+        TOO_MANY_REQUESTS = 429;
+        INTERNAL_SERVER_ERROR = 500;
+        SERVICE_UNAVAILABLE = 503;    
+    }
+
+    string request_id = 1;
+    Status status_code = 10;
+    optional string status_desc = 11;
+    uint32 relay_peer_count = 12;
```

Comment:
What if instead of sending back this number. We require the sender to specify the minimum # of peers sent to that they feel is appropriate?

Reply:
IMO, the `lightPush` should not care about the `relay` connected peers. We can simply return an "LOW_NUMBER_OF_PEERS" error which will make the lightpush client to just try with another relay peer. The current number of relay peers can be returned back within the `status_desc` field

Reply:
I think it makes sense for the service node to return the number of peers and the client to decide what to do with this information.

Reply:
I would close this as is here, returning exact peer count the message had been relayed to, gives the chance and flexibility to the client or SDK to decide its reactions. Other options would make the protocol just even fluid and harder to implement properly for other use cases.

Reply:
@NagyZoltanPeter - I think it would be interesting to make it clear what the `relay_peer_count` means: whether it represents the actual number of connected nodes or the number of nodes that form the mesh. I'm happy with the decision but we might need to add a small comment somewhere to clarify that (even a comment next to this definition would be great :) )

( cc @jm-clius )